### PR TITLE
Fwa 006 horizontal list weather by hour

### PR DESCRIPTION
**Pull request by heupotato**

### CODE REVIEW

File: Flutter_Weather/lib/models/weather_extension.dart [hunk 1]
```diff
@@ -54,22 +58,26 @@ class DayWeather{
...
 
   int get upperLimitTemp
     => weathers.map((hourData) => hourData.temp2m).toList().reduce(max);
 
   int get lowerLimitTemp
     => weathers.map((hourData) => hourData.temp2m).toList().reduce(min);
+
+  @override
+  String toString() {
+    super.toString();
```
Comment (lngyeen) [inline Flutter_Weather/lib/models/weather_extension.dart]: The call to the super class is meaningless in this case. Remove super.toString();

File: Flutter_Weather/lib/widgets/weather_day_detail_widget.dart [hunk 2]
```diff
@@ -15,40 +18,46 @@ class WeatherDayDetail extends StatefulWidget {
...
     }
     return _weatherInfoDay;
   }
 
-  Container _weatherInfoHour(WeatherData hourData, int initTime) {
-    String currentTime = (hourData.timepoint + initTime > 12) ?
-    (hourData.timepoint + initTime - 12).toString() + "PM"
-        : (hourData.timepoint + initTime).toString() + "AM";
+  Container _weatherInfoHour(WeatherData hourData, DateTime initDate) {
+    String currentTime = (hourData.hour(initDate) > 12) ?
```
Comment (lngyeen) [inline Flutter_Weather/lib/widgets/weather_day_detail_widget.dart]: Add a property in WeatherItem class extension called 'localTime' which will return a DateTime object.
Use DateFormatter with a custom format 'hh:mm a' to show 'AM/PM' time.

File: Flutter_Weather/lib/models/weather_extension.dart [hunk 3]
```diff
@@ -54,22 +58,26 @@ class DayWeather{
...
-      message: e.toString());
-    }
+  WeatherData get weatherNow{
+    DateTime now = DateTime.now().toLocal();
+    WeatherData weatherNow = weathers.firstWhere((weatherData) =>
+      now.difference(initDate.add(Duration(hours: weatherData.timepoint)).toLocal()).inMinutes <= 90);
+    print(now);
+    DateTime chosenDate = initDate.add(Duration(hours: weatherNow.timepoint));
+    print(chosenDate);
+    print(now.difference(chosenDate).inHours);
```
Comment (lngyeen) [inline Flutter_Weather/lib/models/weather_extension.dart]: Remove all print methods.

**Review by lngyeen — CHANGES_REQUESTED** [review]

File: Flutter_Weather/lib/widgets/weather_day_detail_widget.dart [hunk 4]
```diff
@@ -15,41 +18,49 @@ class WeatherDayDetail extends StatefulWidget {
...
+    return Padding(
+        padding: EdgeInsets.fromLTRB(0,0,0, 0),
+        child: SizedBox(
+          height: 300,
+          child: ListView.builder(
+            itemCount: _allAvailableDays.length,
+            itemBuilder: (context, index){
+              DayWeather dayWeather = _allAvailableDays[index];
+              return
+                WeatherDayDetailList(weatherInfoDay: _weatherInfoDay(dayWeather, _initDate), weekDay: dayWeather.weekDay,);
```
Comment (lngyeen) [inline Flutter_Weather/lib/widgets/weather_day_detail_widget.dart]: WeatherDayDetailList should have only one param in constructor: DayWeather dayWeather.

File: Flutter_Weather/lib/widgets/weather_day_detail_widget.dart [hunk 5]
```diff
@@ -15,41 +18,49 @@ class WeatherDayDetail extends StatefulWidget {
...
+      )
+    ));
   }
 
   WeatherTypeIcon _weatherIcon(String weather) {
     return WeatherTypeIcon(weather: weather);
   }
 
-  List<Widget> _weatherInfoDay(DayWeather dayData, int initTime) {
+  List<Widget> _weatherInfoDay(DayWeather dayData, DateTime initDate) {
```
Comment (lngyeen) [inline Flutter_Weather/lib/widgets/weather_day_detail_widget.dart]: This function was used to build layout for another widget (WeatherDayDetailList). Should move it inside WeatherDayDetailList class.

File: Flutter_Weather/lib/widgets/weather_day_detail_widget.dart [hunk 6]
```diff
@@ -15,41 +18,49 @@ class WeatherDayDetail extends StatefulWidget {
...
     List<Widget> _weatherInfoDay = [];
-    for (int i = 0; i < 10; i++) {
+    for (int i = 1; i < dayData.weathers.length*2; i++) {
```
Comment (lngyeen) [inline Flutter_Weather/lib/widgets/weather_day_detail_widget.dart]: Use ListView.separated instead, you don't have to use for loop with double of items.

**Review by lngyeen — CHANGES_REQUESTED** [review]

File: Flutter_Weather/lib/widgets/weather_day_detail_widget.dart (hunk 5 above)
Comment (heupotato) [reply Flutter_Weather/lib/widgets/weather_day_detail_widget.dart]: I moved it into separate class to reuse later (it's necessary for task 7)

File: Flutter_Weather/lib/widgets/weather_info_day.dart [hunk 7]
```diff
@@ -0,0 +1,70 @@
+import 'package:flutter/cupertino.dart';
+import 'package:flutter/material.dart';
+import 'package:flutter_weather/models/index.dart';
+import 'package:flutter_weather/models/weather_extension.dart';
+import 'package:flutter_weather/packages/dafluta/dafluta.dart';
+import 'package:flutter_weather/widgets/icons/weather_type_icon.dart';
+
+class WeatherInfoDay{
```
Comment (lngyeen) [inline Flutter_Weather/lib/widgets/weather_info_day.dart]: Why this class only have static functions?
This should be a widget class. Constructor will have tow params WeatherInfoDay({required this.dayData, required this.initDate}).

**Review by lngyeen — CHANGES_REQUESTED** [review]

**Pull request merged**